### PR TITLE
[FBcode->GH] Fix accimage tests

### PR DESCRIPTION
**Pull request by NicolasHug**
Should fix https://github.com/pytorch/vision/issues/5541

### CODE REVIEW

File: torchvision/transforms/functional_pil.py
Comment (prabhat00155) [inline torchvision/transforms/functional_pil.py:48]: This gives `AttributeError: channels`.

**Review by prabhat00155 — CHANGES_REQUESTED** [review]
Thanks @NicolasHug, this doesn't seem to work. Blocking accidental merge.

File: torchvision/transforms/functional_pil.py
Comment (prabhat00155) [reply torchvision/transforms/functional_pil.py:48]: I am running the command mentioned in the diff on my devserver.

**Comment by datumbox** [comment]
@NicolasHug @prabhat00155 Thanks for investigating.

I agree with your comments on the issues that we should consider deprecating `accimage`. Until then we can circumvent the problem with 2 ways:
- Restore the previous inaccurate `channels = 1 if img.mode == "L" else 3`. This is what we had [before](https://github.com/pytorch/vision/pull/5487/files#diff-efb615f437f0b00ef2ed8abf2e0eda487cf609899a115bf3fa4691100db581afL33) but it's obviously wrong.
- Replace it with the inefficient `np.array(img).shape[2]`. Given Nicolas' patch it should only execute when `getbands()` is not available.

I've pushed the changes directly to the @NicolasHug branch, I hope you don't mind. I'll leave it to you to merge this or revert the change. Stamping to unblock.

**Review by datumbox — APPROVED** [review]

File: torchvision/transforms/functional_pil.py
Comment (NicolasHug) [reply torchvision/transforms/functional_pil.py]: Using `conda install -c conda-forge accimage` from the installation instructions of accimage: https://github.com/pytorch/accimage#installation

**Comment by NicolasHug** [comment]
>  Until then we can circumvent the problem with 2 ways:
> I've pushed the changes directly to the @NicolasHug branch,

Thanks @datumbox , did you check that the proposed solution properly fixes the internal tests? Otherwise we shouldn't merge it

**Comment by datumbox** [comment]
I've checked the internal tests and to me `.channels` works fine. So I restored your approach as it's better:


```
buck test @mode/dev //pytorch/vision/test:torchvision_transforms -- --exact 'pytorch/vision/test:torchvision_transforms - test_transforms.py::TestAccImage::test_accimage_crop'
 ....
    ✓ ListingSuccess: pytorch/vision/test:torchvision_transforms : 1310 tests discovered (22.521)
    ✓ Pass: pytorch/vision/test:torchvision_transforms - test_transforms.py::TestAccImage::test_accimage_crop (0.266)
    ✓ Pass: pytorch/vision/test:torchvision_transforms - main (19.158)
Summary
  Pass: 2
  ListingSuccess: 1
```

**Review by prabhat00155 — APPROVED** [review]

**Pull request merged**
